site/en/reference/cli/concepts/dataset.md

```diff
@@ -0,0 +1,189 @@
+
+For a list of `dataset` commands, type `help`:
+
+```
+> dataset help
+help
+active
+activetimestamp
+channel
+channelmask
+clear
+commit
+delay
+extpanid
+init
+meshlocalprefix
+mgmtgetcommand
+mgmtsetcommand
+networkkey
+networkname
+panid
+pending
+pendingtimestamp
+pskc
+securitypolicy
+Done
+```
+
+## Argument mappings
+
+### Security Policy
+
+Security Policy commands use argument mappings to get and set
+[otSecurityPolicy](https://openthread.io/reference/struct/ot-security-policy)
+members. For example, `dataset active`:
+
+```
+> dataset active
+Active Timestamp: 1
+Channel: 13
+Channel Mask: 0x07fff800
+Ext PAN ID: d63e8e3e495ebbc3
+Mesh Local Prefix: fd3d:b50b:f96d:722d::/64
+Network Key: dfd34f0f05cad978ec4e32b0413038ff
+Network Name: OpenThread-8f28
+PAN ID: 0x8f28
+PSKc: c23a76e98f1a6483639b1ac1271e2e27
+Security Policy: 0, onrcb
+Done
+```
+
+In this example, `Security Policy: 0` indicates [mRotationTime](https://openthread.io/reference/struct/ot-security-policy#mrotationtime).
+
+`o`, `n`, `r`, `c`, `b`, `C`, `e`, `p`, `R` values map to
+[mObtainNetworkKeyEnabled](https://openthread.io/reference/struct/ot-security-policy#mobtainnetworkkeyenabled),
+[mNativeCommissioningEnabled](https://openthread.io/reference/struct/ot-security-policy#mnativecommissioningenabled),
+[mRoutersEnabled](https://openthread.io/reference/struct/ot-security-policy#mroutersenabled),
+[mExternalCommissioningEnabled](https://openthread.io/reference/struct/ot-security-policy#mexternalcommissioningenabled),
+[mBeaconsEnabled](https://openthread.io/reference/struct/ot-security-policy#mbeaconsenabled),[mCommercialCommissioningEnabled](https://openthread.io/reference/struct/ot-security-policy#mcommercialcommissioningenabled),
+[mAutonomousEnrollmentEnabled](https://openthread.io/reference/struct/ot-security-policy#mautonomousenrollmentenabled),
+[mNetworkKeyProvisioningEnabled](https://openthread.io/reference/struct/ot-security-policy#mnetworkkeyprovisioningenabled),
+[mNonCcmRoutersEnabled](https://openthread.io/reference/struct/ot-security-policy#mnonccmroutersenabled)
+respectively.
+
+The `dataset securitypolicy` get and set commands also use the same argument
+mappings, for example setting the `securitypolicy` and passing `o`, `n`, `r`,
+and `c`:
+
+```
+> dataset securitypolicy 672 onrc
+Done
+```
+
+### Dataset components and `mgmt` commands
+
+Along with other parameters, the `mgmtgetcommand` and `mgmtsetcommand`
+for Active and Pending Datasets allow you to get and set any combination
+of [otOperationalDatasetComponents](https://openthread.io/reference/struct/ot-operational-dataset-components):
+
+*   `activetimestamp`
+*   `pendingtimestamp`
+*   `networkkey`
+*   `networkname`
+*   `extpanid`
+*   `localprefix`
+*   `delaytimer`
+*   `panid`
+*   `channel`
+*   `securitypolicy`
+
+For the `mgmtgetcommand`, you can specify these components in any order to get
+the corresponding values. Optionally, you can also pass `-x` to use a hex
```

Comment:
Should we include an example that uses -x with a TLV?  How would those be used in the field?

Reply:
Hi @abtink - could you pls help with this? Thanks!

Reply:
Yes, I think we should provide an example of `-x`. For me, this is the easiest way to fully configure a device so that it is a full member of the Thread network. It basically allows copy-pasting a single hex string.

On existing device:
```
> dataset active -x
0e080000000000010000000300001035060004001fffe00208e227ac6a7f24052f0708fdb753eb517cb4d3051062b2442a928d9ea3b947a1618fc4085a030f4f70656e5468726561642d393837330102987304105330d857354330133c05e1fd7ae81a910c0402a0f7f8
Done
```

On new device:
```
> dataset set active 0e080000000000010000000300001035060004001fffe00208e227ac6a7f24052f0708fdb753eb517cb4d3051062b2442a928d9ea3b947a1618fc4085a030f4f70656e5468726561642d393837330102987304105330d857354330133c05e1fd7ae81a910c0402a0f7f8
Done
```

Reply:
Thanks - I'll add this to the **Active** section. It might also help to have an example for the `mgmtgetcommand` and `mgmtsetcommand`...

Reply:
I think the `mgmtgetcommand` and `mgmtsetcommand` will require more extensive documentation to better describe when/how to use them. Currently, they are primarily used for testing. For example, `mgmtgetcommand` does not actually print the values that were received. Instead, `mgmtgetcommand` is currently used to generate packets that allow a test harness to observe the resulting behavior.

Reply:
Maybe we can do the examples for the `mgmt` commands in a separate issue, so we don't hold up this PR.  Having the `dataset active -x` one should be okay for now?

Reply:
> Maybe we can do the examples for the `mgmt` commands in a separate issue, so we don't hold up this PR. Having the `dataset active -x` one should be okay for now?

Yes, I agree.

Reply:
Thanks for the feedback - I made some updates per this + offline discussions. I also updated the openthread PR.